x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connector_detail/native_connector_configuration.tsx

```diff
@@ -169,23 +171,27 @@ export const NativeConnectorConfiguration: React.FC = () => {
                   ),
                   titleSize: 'xs',
                 },
-                {
-                  children: (
-                    <ApiKeyConfig
-                      indexName={connector.index_name || ''}
-                      hasApiKey={hasApiKey}
-                      isNative
-                    />
-                  ),
-                  status: hasApiKey ? 'complete' : 'incomplete',
-                  title: i18n.translate(
-                    'xpack.enterpriseSearch.content.indices.configurationConnector.nativeConnector.steps.manageApiKeyTitle',
-                    {
-                      defaultMessage: 'Manage API key',
-                    }
-                  ),
-                  titleSize: 'xs',
-                },
+                ...(apiKeysEnabled
+                  ? [
+                      {
+                        children: (
+                          <ApiKeyConfig
+                            indexName={connector.index_name || ''}
+                            hasApiKey={hasApiKey}
+                            isNative
+                          />
+                        ),
+                        status: hasApiKey ? 'complete' : 'incomplete',
+                        title: i18n.translate(
+                          'xpack.enterpriseSearch.content.indices.configurationConnector.nativeConnector.steps.manageApiKeyTitle',
+                          {
+                            defaultMessage: 'Manage API key',
+                          }
+                        ),
+                        titleSize: 'xs',
+                      } as EuiContainedStepProps,
```

Comment:
One quick question after my approval: why do you need to typecast this? What is the compiler missing?

Reply:
I'll merge for now but if there's a better approach I'll implement it in another PR, let me know

Reply:
Given the error a better fix for this would be to `as` just the `status` value.

`status: (hasApiKey ? 'complete' : 'incomplete') as EuiStepStatus,` should work

Reply:
Another possible option, but not as good, would be to `as` just the strings:

`status: hasApiKey ? 'complete' as 'complete' : 'incomplete' as 'incomplete',`

This will ensure typescript sees them as their strict values and not `string`.